Update scoped-transitions.md. (#236)
WICG/view-transitions: scoped-transitions.md

```
div.scope-root
└─ ::view-transition
   ├─ ::view-transition-group(outer)
   │  └─ ::view-transition-image-pair(outer)
   │     ├─ ::view-transition-old(outer)
   │     └─ ::view-transition-new(outer)
   └─ ::view-transition-group(inner)
      └─ ::view-transition-image-pair(inner)
         ├─ ::view-transition-old(inner)
         └─ ::view-transition-new(inner)
```

## Algorithm

The steps for a scoped view transition are based on the
[steps for a document view transition](https://drafts.csswg.org/css-view-transitions-1/#lifecycle)
with appropriate modifications.  At a high level:

1. Create the [`ViewTransition`](https://drafts.csswg.org/css-view-transitions-1/#viewtransition) object.

2. At the next rendering opportunity, capture the painted output of each tagged
   element under the scoped transition root, and create the pseudo-element tree
   with `::view-transition-old` pseudo-elements.  A tagged element's geometry
   information is computed relative to the scoped transition root.

3. Invoke the callback passed to `startViewTransition`.

4. Create the `::view-transition-new` pseudo-elements and set up the default
   animations.

5. Run the animations.

> 6. Clean up by destorying the pseudo-element tree.

Reviewer: Typo:

```suggestion
6. Clean up by destroying the pseudo-element tree.
```

Between steps 2 and 4, we need to pause the rendering of the scoped transition

> root's subtree so that the developer can update it without presenting those

Reviewer: I think you mean the UA?

```suggestion
root's subtree so that the user-agent can update it without presenting those
```

Author: Actually I meant the developer since the updates occur in the DOM callback.

Reviewer: Well yeah, it’s the UA doing the update by invoking the callback that the author specified.

“…  so that the developer can update it …” reads to my as the developer manually doing some DOM manipulations in between those steps.

Maybe rephrase to the following then:

```suggestion
root's subtree so that the user-agent can execute the callback without presenting those
```

?

Author: Tweaked the wording, PTAL

Reviewer: Perfect!

updates to the user.

## Constraints

> Scoped view transitions imposes certain constraints:

Reviewer: Typo:

```suggestion
Scoped view transitions impose certain constraints:
```

* There shouldn't be more than one active transition running on the same scoped
  transition root. If a new transition is started on the same element, we
  should cancel the old one.

* A tagged element cannot participate (by generating a `::view-transition-group`)
  in more than one active transition at the same time. If a new transition is
  started which would trigger this situation, we should cancel the old one.

* The scoped transition root must have `contain: layout`. This ensures that it

>   generates a [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context)

Reviewer: Best to use unlocalized MDN URLs:

```suggestion
  generates a [stacking context](https://developer.mozilla.org/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context)
```
